### PR TITLE
Extend schemas endpoint

### LINKED ISSUE #1449: Extend schemas endpoint
### Description

Extend the following endpoint to include the response from: `auth_getMultiSigRegMsgSchema` endpoint

```
HTTP GET /api/v3/schemas
RPC  get.schemas
```

### Acceptance Criteria

- The above endpoints are up-to-date
- Response matches as shown below
  - `messages` property should always be available
  - Include the specified schema iff both the `auth` module and the `auth_getMultiSigRegMsgSchema` endpoint are registered
- Gateway source mappings are up-to-date
- The test schema and related integration tests are up-to-date
- Swagger is up-to-date
  - Ensure the logic behind the structure and usage is properly explained with an example

### Additional Information

Response: *200 OK*
```jsonc
{
    "data": {
        "block": { ... },
        "header": { ... },
        "asset": { ... },
        "transaction": { ... },
        "event":  { ... },
        "events": [{ ... }],
        "assets": [{ ... }],
        "commands": [{ ... }],
        "messages": [  // <--- Add the following
          {
            "moduleCommand": "auth:registerMultisignature",
            "param": "signatures",
            "schema": { ... },
          },
        ],
    },
    "meta": {}
}
```

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1449 

### How was it solved?

- [x] The above endpoints are up-to-date
- Response matches as shown below
    - [x] messages property should always be available
    - [x] Include the specified schema iff both the auth module and the auth_getMultiSigRegMsgSchema endpoint are registered
- [x] Gateway source mappings are up-to-date
- [x] The test schema and related integration tests are up-to-date
    - [x] Update header schema
- [x] Swagger is up-to-date
    - [x] Ensure the logic behind the structure and usage is properly explained with an example
    - [x] Add definition for header schema
- [x] Add standardEvent schema
### How was it tested?
-  [x] Local

### CODE REVIEW

File: tests/schemas/api_v3/allSchemas.schema.js [hunk 1]
```diff
@@ -59,6 +65,7 @@ const allSchemasSchema = {
 	events: Joi.array().items(eventsSchema).required(),
 	assets: Joi.array().items(assetsSchema).required(),
 	commands: Joi.array().items(commandsParamsSchemasSchema).required(),
+	messages: Joi.array().items(messagesSchema).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/allSchemas.schema.js:68]:
```suggestion
	messages: Joi.array().items(messagesSchema).min(0).required(),
```

File: services/blockchain-indexer/shared/dataService/business/schemas.js [hunk 2]
```diff
@@ -13,24 +13,46 @@
...
+const getAuthMultiSigRegMsgSchema = async () => {
+	const registeredModules = await getRegisteredModules();
+	const isAuthModuleRegistered = registeredModules.includes(MODULE.AUTH);
+
+	if (isAuthModuleRegistered) {
+		const authMultiSigRegMsgSchema = await requestConnector('getAuthMultiSigRegMsgSchema');
+		return {
+			moduleCommand: `${MODULE.AUTH}:registerMultisignature`,
+			param: 'signatures',
+			...authMultiSigRegMsgSchema, // schema
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/dataService/business/schemas.js:30]: The actual schema should be identified by the `schema` property within a messageSchema.
```suggestion
			schema: authMultiSigRegMsgSchema,
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 3]
```diff
@@ -203,6 +203,299 @@
             }
         }
     },
+    "headerSchema": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:206]: The swagger definition identifiers must always be defined in `PascalCase`. Please ensure to update the existing ones as and when we come across them.
```suggestion
    "HeaderSchema": {
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 4]
```diff
@@ -203,6 +203,299 @@
...
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    },
     "assetSchema": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:499]: Please make necessary changes everywhere else.
```suggestion
    "AssetSchema": {
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 5]
```diff
@@ -762,6 +1055,174 @@
             }
         }
     },
+    "messageSchema": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1058]:
```suggestion
    "MessageSchema": {
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 6]
```diff
@@ -762,6 +1055,174 @@
...
+                "type": "string",
+                "example": "auth:registerMultisignature",
+                "description": "Combination of transaction moduleName:commandName.\n"
+            },
+            "param": {
+                "type": "string",
+                "example": "signatures"
+            },
+            "schema": {
+                "type": "object",
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1076]: If possible, please add the `"required": []` property as well.

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 7]
```diff
@@ -762,6 +1055,174 @@
...
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    },
     "schemaWithEnvelope": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1226]:
```suggestion
    "SchemaWithEnvelope": {
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 8]
```diff
@@ -203,6 +203,299 @@
...
+            "schema"
+        ],
+        "properties": {
+            "schema": {
+                "type": "object",
+                "properties": {
+                    "$id": {
+                        "type": "string",
+                        "example": "/block/header/3",
+                        "description": "Unique identifier of the block schema\n"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:218]:
```suggestion
                        "description": "Unique identifier of the block schema."
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 9]
```diff
@@ -203,6 +203,299 @@
...
+                    },
+                    "type": {
+                        "type": "string",
+                        "example": "object",
+                        "description": "type of schema"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:223]:
```suggestion
                        "description": "Type of schema."
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 10]
```diff
@@ -762,6 +1055,174 @@
...
+        "required": [
+            "moduleCommand",
+            "param",
+            "schema"
+        ],
+        "properties": {
+            "moduleCommand": {
+                "type": "string",
+                "example": "auth:registerMultisignature",
+                "description": "Combination of transaction moduleName:commandName.\n"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1069]:
```suggestion
                "description": "Combination of transaction moduleName:commandName."
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 11]
```diff
@@ -762,6 +1055,174 @@
...
+            },
+            "schema": {
+                "type": "object",
+                "properties": {
+                    "$id": {
+                        "type": "string",
+                        "format": "id",
+                        "minLength": 1,
+                        "example": "/auth/command/regMultisigMsg",
+                        "description": "Unique identifier of the message schema\n"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1083]:
```suggestion
                        "description": "Unique identifier of the message schema."
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 12]
```diff
@@ -762,6 +1055,174 @@
...
+                    },
+                    "type": {
+                        "type": "string",
+                        "example": "object",
+                        "description": "type of schema"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1088]:
```suggestion
                        "description": "Type of schema."
```

File: services/gateway/apis/http-version3/swagger/definitions/schemas.json [hunk 13]
```diff
@@ -762,6 +1055,174 @@
...
+        ],
+        "properties": {
+            "moduleCommand": {
+                "type": "string",
+                "example": "auth:registerMultisignature",
+                "description": "Combination of transaction moduleName:commandName.\n"
+            },
+            "param": {
+                "type": "string",
+                "example": "signatures"
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/definitions/schemas.json:1073]:
```suggestion
                "example": "signatures",
                "description": "The property within the params of the corresponding moduleCommand type transaction for which the corresponding schema is to be used to serialize the information.\nExample: The schema with `\"$id\": \"/auth/command/regMultisigMsg\"` is to be used to serialize the `signatures` data within the transaction params for the `auth:registerMultisignature` command."
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-indexer/shared/dataService/business/schemas.js (hunk 2 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/schemas.js:30]: SDK returns following structure:

```
{
    schema: {
        ...
    }
}
```

Comment (sameersubudhi) [reply services/blockchain-indexer/shared/dataService/business/schemas.js:30]: Please rename the variable/refactor the code to make it more intuitive. `authMultiSigRegMsgSchema` is misleading.

Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/schemas.js]: Refactored.

File: services/blockchain-connector/shared/sdk/auth.js [hunk 14]
```diff
@@ -31,6 +31,19 @@ const getAuthAccount = async (address) => {
...
 
+const getAuthMultiSigRegMsgSchema = async () => {
+	try {
+		const multiSigRegMsgSchema = await invokeEndpoint('auth_getMultiSigRegMsgSchema');
+		return multiSigRegMsgSchema;
+	} catch (err) {
+		if (err.message.includes(timeoutMessage)) {
+			throw new TimeoutException('Request timed out when calling \'getMultiSigRegMsgSchema\'.');
+		}
+		throw err;
```
Comment (nagdahimanshu) [inline services/blockchain-connector/shared/sdk/auth.js:42]: We can add warn message to make it consistent
```suggestion
		if (err.message.includes(timeoutMessage)) {
			throw new TimeoutException('Request timed out when calling \'getMultiSigRegMsgSchema\'.');
		}
		logger.warn(`Error returned when invoking 'auth_getMultiSigRegMsgSchema'.\n${err.stack}`);
		throw err;
```

File: services/blockchain-indexer/shared/dataService/business/schemas.js [hunk 15]
```diff
@@ -13,24 +13,46 @@
  * Removal or modification of this copyright notice is prohibited.
  *
  */
+const { getRegisteredModules, MODULE } = require('../../constants');
 const { requestConnector } = require('../../utils/request');
 
 let allSchemas;
 
+const getAuthMultiSigRegMsgSchema = async () => {
```
Comment (nagdahimanshu) [inline services/blockchain-indexer/shared/dataService/business/schemas.js:21]: I think its better to move this method `getAuthMultiSigRegMsgSchema` to `services/blockchain-indexer/shared/dataService/business/auth` as it is relevant to auth module.

File: services/blockchain-connector/shared/sdk/auth.js (hunk 14 above)
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/auth.js]: Applied.

File: services/blockchain-indexer/shared/dataService/business/schemas.js (hunk 15 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/schemas.js]: Applied.

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**